src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -522,6 +530,50 @@ public static int[] GetBits(decimal d)
             return new int[] { d.lo, d.mid, d.hi, d.flags };
         }
 
+        /// <summary>
+        /// Converts the value of a specified instance of <see cref="decimal"/> to its equivalent binary representation.
+        /// </summary>
+        /// <param name="d">The value to convert.</param>
+        /// <param name="destination">The span into which to store the four-integer binary representation.</param>
+        /// <returns>Four, the number of integers in the binary representation.</returns>
+        /// <exception cref="ArgumentException">The destination span was not long enough to store the binary representation.</exception>
+        public static int GetBits(decimal d, Span<int> destination)
+        {
+            if ((uint)destination.Length <= 3)
```

Comment:
why `<= 3` rather than `< 4`?

Reply:
Because for whatever reason the JIT removes the bounds checks with the former but not with the latter.